### ISSUE TITLE
VSCE: Automate rendering of user's color theme 

### ISSUE THREAD

**Issue by jjinnii** [sourcegraph/sourcegraph#29704]
The expected behavior of the VSCE is to pull in a user's color theme palette for all UI elements, including text color, background color for containers, icon colors, etc. This allows the extension to feel more "native' and integrated to VS Code extension and prevents scenarios in which there is a harsh contrast btw a user's default color palette and switching to our extension. 

It's important that we check with multiple color palettes and ensure that we are pulling in these color themes in an effective and accessible way -- this will define at a glance how the extension looks to a user.

**Comment by jjinnii** [comment]
@tjkandala Just wanted to follow up on your question from the planning meeting. This issue covers essentially spans all colors on the VSCE, which includes syntax highlighting. At the moment, it seems to be a hybrid of pulling in a user's color theme and Sourcegraph colors (definitely my bad here, my designs promoted this). I think the most natural state for the extension would be to pull in the user's color theme in to all elements: syntax highlighting, typography, background colors, etc. 
<img width="1089" alt="Screen Shot 2022-01-13 at 1 23 22 PM" src="https://user-images.githubusercontent.com/87138876/149396285-524e3158-00a5-4c0c-b8f2-23e134b716b4.png">
<img width="1415" alt="Screen Shot 2022-01-13 at 1 23 08 PM" src="https://user-images.githubusercontent.com/87138876/149396289-00567eb7-e5af-4955-8bbe-53ca644db2d4.png">

Please let me know if you'd like me to clarify further!

**Comment by tjkandala** [comment]
@jjinnii Got it! I believe this will basically entail finding VS Code CSS variables to use in place of [our highlighting styles](https://sourcegraph.com/github.com/sourcegraph/sourcegraph@vsce/vscode-open-in-browser/-/blob/client/branded/src/global-styles/highlight.scss)

**Comment by jjinnii** [comment]
@tjkandala Perfect! Do let me know when this issue is ready for review -- I'd like to test with a few different themes, just to make sure that for the most popular themes in VS Code at least, our extension looks sleek and polished with how/what CSS variables we're pulling in for the different components :)

**Comment by jjinnii** [comment]
I know we're still working on this -- but just for reference, this is what shows up for me on VS Code's default dark theme with the new version from the .vsix file. Let's make sure "error red" isn't the default color for highlighting 😆 

<img width="1440" alt="Screen Shot 2022-01-24 at 3 04 29 PM" src="https://user-images.githubusercontent.com/87138876/150868945-8707bdfc-7112-460f-a298-f73d3d0f76ea.png">

**Comment by abeatrix** [comment]
Updated to use `--vscode-editor-selectionBackground` 
![image](https://user-images.githubusercontent.com/68532117/150887931-01e8488c-b22a-4786-bf0d-c19587a94ece.png)

**Comment by jjinnii** [comment]
@abeatrix  That looks a lot better! Could you please test out a few other default themes with your latest version to make sure nothing stands out? Thank you!

**Comment by abeatrix** [comment]
For syntax highlighting, it looks like currently access to VS Code built-in highlighting color is not available: https://github.com/microsoft/vscode/issues/56356
https://github.com/microsoft/vscode/issues/91375
Will keep looking to see if there is a workaround

**Comment by abeatrix** [comment]
This is currently blocked by the limitation of the VS Code API. A potential temporary workaround I can think of would be adding an option for users to provide the extension with their color theme.

**Comment by jjinnii** [comment]
Could I get some clarification on what "providing the extension with their color theme" means? Namely: 
* How would a user share/provide their color theme?
* What would be do with that information to support their color theme? And are there additional actions a user needs to take?

**Comment by abeatrix** [comment]
> How would a user share/provide their color theme?
The user would first need to generate a color theme file using the `Developer: Generate Color Theme from Current Settings` command from their VS Code `Command Palette`, and then from there I was thinking if we can write a program that would take the newly-generated color theme from the user and extract the color for tokens use, and save it as a new sourcegraph setting that allows the extension to use it (or even in cloud for future)

**Comment by abeatrix** [comment]
Moving this to `Blocked` because as mentioned above, this is blocked by the limitation of the VS Code API since theme and token colors are not accessible via the API: https://github.com/sourcegraph/sourcegraph/issues/29704#issuecomment-1021569757

**Comment by jjinnii** [comment]
Thanks @abeatrix! Just to refresh my memory:

- What we have implemented now: Coverage for most highly-used VS Code theme palettes, which required manual hard-coding of CSS variables. 
- What this issue aims to do: A solution that automatically mirrors a user's VS Code theme palette, without having to hard code any CSS. 

And we are blocked because it's not possible for us to pull in theme colors through the API, and the only solution would be for a user to follow some manual steps and save as a Sourcegraph extension setting? 

If above is correct, what would you say our coverage of themes on VS Code is? e.g. we have the top 20 themes hard coded.

**Comment by jjinnii** [comment]
Thanks for the clarification @abeatrix ! I changed the name of the issue to be a bit clearer. If we no see no feasible technical way to implement this, I think we can close this issue. 😄 We've received no feedback yet (besides positive ones!) on user color themes, so I think we can create a new issue in the future when we know of any specific user problem.
